virtcontainers/cc_proxy.go

```diff
@@ -13,7 +13,7 @@ type ccProxy struct {
 }
 
 // start is the proxy start implementation for ccProxy.
-func (p *ccProxy) start(sandbox Sandbox, params proxyParams) (int, string, error) {
+func (p *ccProxy) start(sandbox *Sandbox, params proxyParams) (int, string, error) {
```

Comment:
Don't we now need check on all these params now they are pointers?:

```
var errNeedSandbox = errors.New("Sandbox cannot be empty")

    :

func (p *ccProxy) start(sandbox *Sandbox, params proxyParams) (int, string, error) {
    if sandbox == nil {
        return -1, "", errNeedSandbox
    }

    :
}

```

Reply:
These are all internal functions. I think we should trust internal callers because otherwise we do not trust ourselves. That said, even if we do not trust internal callers, the check should be added as a separate PR since it is an existing issue for places where sandbox is already passed as pointer.

Reply:
@jodh-intel Do you want to check every parameter in every function? Since that's what _not trust anything_ means to me. From my point of view, I think we should check all parameters in all of the virtcontainers APIs but that's all. We are already doing this and it ensures that we fail fast.

OTOH, if we do not trust ourselves and check parameter in every function, we waste CPU time and memory on checking the same thing over and over again. Please create an issue for it if you think that's really necessary. I don't mind having such a discussion but it does not need to block this PR since it's a different/existing issue.

Reply:
A few examples of where `*Sandbox` is not checked before being dereferenced in master:
- https://github.com/kata-containers/runtime/blob/master/virtcontainers/kata_agent.go#L132
- https://github.com/kata-containers/runtime/blob/master/virtcontainers/kata_agent.go#L183
- https://github.com/kata-containers/runtime/blob/master/virtcontainers/shim.go#L175
- https://github.com/kata-containers/runtime/blob/master/virtcontainers/container.go#L404

There a more of them for sure. And if we want to take precaution in all these functions, it should also apply to other parameters as well, not just the sandbox pointers.